# Most frequent used Apps in Android and Apple Stores

In this project, we will examine the datasets of the Android and IOS apps stores to analyze the most installed and used apps by the users of both stores. The datasets for both google play and apple stores are available in the `Profitable Apps for Android and IOS stores` repository and are extracted from:
   * [apple store data](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)
   * [google play store data](https://www.kaggle.com/lava18/google-play-store-apps).

Our job is to enable our team of developers to make data-driven decisions with respect to the kind of apps they build. At our company, we only build apps that are free to download & install and that are directed toward an English-speaking audience.

### Uploading the datasets

In [1]:
# opening files
opened_file_1 = open('AppleStore.csv', encoding='utf8')
opened_file_2 = open('googleplaystore.csv', encoding='utf8')

#reading files
from csv import reader
read_file_1 = reader(opened_file_1)
read_file_2 = reader(opened_file_2)

#transforming both files for apple & google play stores into list of lists
ios_data = list(read_file_1)
ios_header = ios_data[0]
ios_data = ios_data[1:]

android = list(read_file_2)
android_header = android[0]
android = android[1:]

### Exploring the datasets

To explore the datasets, we will write a function `explore_data` to use it repetitively and to explore datasets in a readable way. 

In [2]:
# defining a function to explore the datasets
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    
    for row in dataset_slice:
        print(row)
        print('\n') # adding a new line after each row
    
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))          

In [3]:
#exploring the ios dataset (including header)
print(ios_header)
print('\n')
explore_data(ios_data, 0, 3, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


Above output shows that Apple store dataset has 7197 apps, 16 columns and the columns that looks interesting includes 'track_name', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'prime_genre'. Details about each column is available [here](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)

In [4]:
#exploring the google play dataset
print(android_header)
print('\n')
explore_data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


We can see that in the google play store dataset, we have 10841 rows with 13 columns and columns that might be useful are: 'App', 'Category', 'Installs', 'Price', 'Genres'. Details for each column is available [here](https://www.kaggle.com/lava18/google-play-store-apps)

### Data cleaning
#### Removal of inaccurate data or duplicate(s)

The Google Play dataset has a dedicated [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion), and we can see that [one of the discussions](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) outlined an error for row 10472. Let's print this row and compare it against the header and another row that is correct.

In [5]:
#examining data for inaccurate value(s)
print(android_header)
print('\n')
print(android[10472])
print('\n')
print(android[0])      # correct row

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


In [6]:
# we'll delete row (indexed as) 10472 as this contains `rating` value as `19` that is beyond
# the maximum rating cap by google, i.e. 5
del android[10472]

In [7]:
#verified the deletion of row
explore_data(android, 10471, 10473)

['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']


['osmino Wi-Fi: free WiFi', 'TOOLS', '4.2', '134203', '4.1M', '10,000,000+', 'Free', '0', 'Everyone', 'Tools', 'August 7, 2018', '6.06.14', '4.4 and up']




In [8]:
#checking android data for duplicate entries:

for app in android:
    name = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


In [9]:
#checking length and names of duplicate apps in android:

duplicate_list = []
unique_list = []

#google play store dataset
for app in android:
    name = app[0]
    if name in unique_list:
        duplicate_list.append(name)
    else:
        unique_list.append(name)

print('Number of Duplicate apps:', len(duplicate_list))
print('\n')
print('Names of Duplicate apps:', duplicate_list[:13])

Number of Duplicate apps: 1181


Names of Duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses']


We have examined that `android` dataset has 1181 duplicate names of apps and therefore, we would remove the duplicates for reliability. Before doing so, we need to determine criterion to remove duplicates instead of doing it randomly, as dataset contains necessary information required for analysis.

In below cell, we printed & examined two cells, the main difference happens on the fourth position of each row, which corresponds to the number of reviews. The different numbers show that the data was collected at different times. We can use this to build a criterion for keeping rows. Therefore, we will create a dictionary where each key is a unique app and value will be maximum number of reviews.

In [10]:
# reviewing the data before determining the criterion
print(android_header)
print('\n')
explore_data(android, 0, 2)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']




In [11]:
# creating a dictionary for the removal of duplicates
reviews_max = {}
for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (name in reviews_max) and (reviews_max[name] < n_reviews):
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

In [12]:
reviews_max

{'Temple Run 2': 8119154.0,
 'du app': 7578.0,
 'Weapon stripping 3D': 23971.0,
 'Learn To Draw Kawaii Characters': 55.0,
 'Beauty Makeup Snappy Collage Photo Editor - Lidow': 420973.0,
 '2017 BN SM Sales Conference': 2.0,
 'Q Cat Live Wallpaper': 4383.0,
 'Six Packs for Man–Body Building with No Equipment': 2880.0,
 'Shadow Fight 2 Special Edition': 10440.0,
 'BN Pro PercentXL-b Neon HD Txt': 45.0,
 'Ah! Coins': 256.0,
 'FH Calculator': 9.0,
 'Wedding Countdown Widget': 7376.0,
 'DT Manager': 9.0,
 'My Airtel-Online Recharge, Pay Bill, Wallet, UPI': 1498393.0,
 'My Calendar - Period Tracker': 156410.0,
 'M-Sight Pro': 30.0,
 'Q-See Plus': 32.0,
 'Meet4U - Chat, Love, Singles!': 40039.0,
 'AK-47: Simulator and Shooting': 345.0,
 'DC Legends: Battle for Justice': 355613.0,
 'YouNow: Live Stream Video Chat': 309872.0,
 'OfficeSuite Pro + PDF (Trial)': 88073.0,
 'Alabama DMV Permit Test - AL': 218.0,
 'Circle ratio': 10562.0,
 'Texas Holdem & Omaha Poker: Pokerist': 187200.0,
 'BN Pro Pla

In [13]:
expected_rows = len(android) - len(duplicate_list)

In [14]:
print(len(reviews_max))
print(len(reviews_max) == expected_rows)

9659
True


We will use above created dictionary to remove duplicates by initiating two empty lists, `android_clean` that will store the new/cleaned apps and `already_added` that will just store the apps. Below, we loop through the android data set, and for every iteration:
- We isolate the name of the app and the number of reviews.
- We add the current row (app) to the android_clean list, and the app name (name) to the already_cleaned list, if:
  - The number of reviews of the current app matches the number of reviews of that app as described in the reviews_max dictionary; and
  - The name of the app is not already in the already_added list. We need to add this additional condition to account for those cases where the highest number of reviews of a duplicate app is the same for more than one entry.

In [15]:
#initiate two empty lists
android_clean = [] 
already_added = [] 

for app in android:
    name = app[0]
    n_reviews = float(app[3])

    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)

In [16]:
# verifying if dataset `android_clean` has 9659 rows
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns: 13


#### Removing non-english apps

Now, we will explore our datasets (ios_data and android) to see if there are any non-english apps and to remove them from datasets.

In [17]:
print(ios_data[813][1])
print(ios_data[6731][1])
print('\n')
print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜


中国語 AQリスニング
لعبة تقدر تربح DZ


We have extracted data of non-english apps from both datasets. Now, we will create a function to determine if character(s) in an app is in ASCII and that each character's corresponding encoding number (in ASCII) is between 0 and 127 as this will help as extracting non-ascii characters. In our function below, we will use built-in `ord()` function to find out the corresponding encoding number of each character.

In [18]:
def is_english(a_string):
    for character in a_string:
        if ord(character) > 127:
            return False
    return True

In [19]:
print(is_english('Instagram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))

True
False


In [20]:
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

print(ord('™'))
print(ord('😜'))

False
False
8482
128540


The function seems to work fine, but some English app names use emojis or other symbols (™, — (em dash), – (en dash), etc.) that fall outside of the ASCII range. Because of this, we'll remove useful apps if we use the function in its current form.

In [21]:
# to minimize loss, we will redefine function to include atmost 3 ASCII characters in an app name
def is_english(a_string):
    non_ascii_charac = 0
    for character in a_string:
        if ord(character) > 127:
            non_ascii_charac += 1
    if non_ascii_charac > 3:
        return False
    return True

In [22]:
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
True


In [23]:
# utilizing the is_english(a_string) function to filter out the non-English apps for both data sets:
ios_english = []
android_english = []
for app in ios_data:
    name = app[1]
    if is_english(name):
        ios_english.append(app)

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)

In [24]:
explore_data(ios_english, 0, 3, True)
print('\n')
explore_data(android_english, 0, 3, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 6183
Number of columns: 16


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Vari

After removal of non-english apps, we can see that we're left with 6183 iOS apps and 9614 Android apps.

#### Isolating the free apps

In [42]:
ios_free_apps = []
android_free_apps = []
for app in ios_english:
    app_price = float(app[4])
    if app_price == 0.0:
        ios_free_apps.append(app)
    
for app in android_english:
    app_price = app[7]
    if app_price is '0':
        android_free_apps.append(app)

In [43]:
print('IOS free apps:', len(ios_free_apps))
print('Android free apps:', len(android_free_apps))

IOS free apps: 3222
Android free apps: 8864


We are left with 3222 ios apps and 8864 android apps for the analysis.

#### Most Common Apps by Genre

Already mentioned in the introduction that our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps. Moreover, our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification.

Furthermore, to minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:
- Build a minimal Android version of the app, and add it to Google Play.
- If the app has a good response from users, we develop it further.
- If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

Let's begin the analysis by getting a sense of what are the most common genres for each market. For this, we'll need to build frequency tables for a few columns in our data sets.

In [36]:
print('Remaining ios_data rows:', len(ios_english_list))
print('\n')
print('Remaining android rows:', len(android_english_list))

Remaining ios_data rows: 6273


Remaining android rows: 10804


In [27]:
# Inspecting both datasets to identify the columns we could use to generate frequency tables and  
# to find out what are the most common genres in each market.

print(ios_header)
print('\n')
explore_data(ios_english, 0, 2)
print('\n')
print(android_header)
print('\n')
explore_data(android_english, 0, 2)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']




['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'A

After examining the above datasets, we will build a frequency table for the `prime_genre column` of the App Store dataset, and the `Genres` and `Category` columns of the Google Play dataset. To achieve this, We'll build two functions which we can use to analyze the frequency tables:
- One function to generate frequency tables that show percentages
- Another function that we can use to display the percentages in a descending order

In [64]:
def freq_table(dataset, index):
    f_table = {}
    f_total = 0
    
    for genre in dataset:
        f_total += 1
        value = genre[index]
        if value in f_table:
            f_table[value] += 1
        else:
            f_table[value] = 1
            
    f_table_percentages = {}
    for key in f_table:
        percentage = (f_table[key] / f_total) * 100
        f_table_percentages[key] = percentage
    return f_table_percentages

In [65]:
def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
        
    sorted_table = sorted(table_display, reverse=True)
    for entry in sorted_table:
        print(entry[1], ':', entry[0])

In [66]:
# examining frequency table for `prime genre` column of apple store dataset
display_table(ios_free_apps, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We can see that among the free English apps, more than a half (58.16%) are games. Entertainment apps are close to 8%, followed by photo and video apps, which are close to 5%. Only 3.66% of the apps are designed for education, followed by social networking apps which amount for 3.29% of the apps in our data set.

The general impression is that App Store (atleast the part containing free English apps) is dominated by apps that are designed for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are more rare.

In [67]:
# examining frequency table for `Category` column of google play store dataset
display_table(android_free_apps, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

On Google Play, there are not that many apps designed for fun, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.)

In [68]:
# examining frequency table for `Genres` column of google play store dataset
display_table(android_free_apps, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The `Genres` column is much more granular (it has more categories).

Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps.

#### Most Popular Apps by Genre on the App Store

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but for the App Store data set this information is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

In [77]:
# Below, we calculate the average number of user ratings per app genre on the App Store:

unique_ios_genre = freq_table(ios_free_apps, -5)

for genre in unique_ios_genre:
    total = 0
    len_genre = 0
    for app in ios_free_apps:
        genre_app = app[-5]
        if genre_app == genre:
            n_user_ratings = float(app[5])
            total += n_user_ratings
            len_genre += 1
    avg_number_ratings = total / len_genre
    print(genre, ':', avg_number_ratings)

News : 21248.023255813954
Games : 22788.6696905016
Travel : 28243.8
Utilities : 18684.456790123455
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Social Networking : 71548.34905660378
Reference : 74942.11111111111
Medical : 612.0
Entertainment : 14029.830708661417
Finance : 31467.944444444445
Business : 7491.117647058823
Photo & Video : 28441.54375
Education : 7003.983050847458
Catalogs : 4004.0
Productivity : 21028.410714285714
Music : 57326.530303030304
Navigation : 86090.33333333333
Food & Drink : 33333.92307692308
Shopping : 26919.690476190477
Lifestyle : 16485.764705882353
Book : 39758.5
Sports : 23008.898550724636


On average, navigation apps have the highest number of user ratings.

#### Most Popular Apps by Genre on Google Play

For the Google Play market, we actually have data about the number of installs, so we should be able to get a clearer picture about genre popularity.

In [81]:
display_table(android_free_apps, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


One issue with the above data is that it is not precise. For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, here we don't need very precise data for our purposes — we only want to get an idea which app genres attract the most users. 
To perform computations, however, we'll need to convert each install number to float — this means that we need to remove the commas and the plus characters.

In [85]:
unique_android_categories = freq_table(android_free_apps, 1)

for category in unique_android_categories:
    total = 0
    len_category = 0
    for app in android_free_apps:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',', '')
            n_installs = n_installs.replace('+', '')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category, ':', avg_n_installs)

EDUCATION : 1833495.145631068
FAMILY : 3695641.8198090694
HEALTH_AND_FITNESS : 4188821.9853479853
VIDEO_PLAYERS : 24727872.452830188
BEAUTY : 513151.88679245283
BUSINESS : 1712290.1474201474
PARENTING : 542603.6206896552
TOOLS : 10801391.298666667
GAME : 15588015.603248259
LIFESTYLE : 1437816.2687861272
SPORTS : 3638640.1428571427
PERSONALIZATION : 5201482.6122448975
LIBRARIES_AND_DEMO : 638503.734939759
FINANCE : 1387692.475609756
COMMUNICATION : 38456119.167247385
PRODUCTIVITY : 16787331.344927534
MEDICAL : 120550.61980830671
EVENTS : 253542.22222222222
COMICS : 817657.2727272727
PHOTOGRAPHY : 17840110.40229885
SOCIAL : 23253652.127118643
MAPS_AND_NAVIGATION : 4056941.7741935486
WEATHER : 5074486.197183099
HOUSE_AND_HOME : 1331540.5616438356
FOOD_AND_DRINK : 1924897.7363636363
TRAVEL_AND_LOCAL : 13984077.710144928
BOOKS_AND_REFERENCE : 8767811.894736841
DATING : 854028.8303030303
NEWS_AND_MAGAZINES : 9549178.467741935
ART_AND_DESIGN : 1986335.0877192982
ENTERTAINMENT : 11640705.88235

On average, communication apps have the most installs: 38,456,119. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [87]:
for app in android_free_apps:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

The books and reference genre looks fairly popular as well, with an average number of installs of 8,767,811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:

In [89]:
for app in android_free_apps:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

In [91]:
for app in android_free_apps:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                            or app[5] == '500,000,000+'
                                            or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


However, it looks like there are only a few very popular apps, so this market still shows potential.

#### Conclusion

We conclude here that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.`